python/cudf/cudf/core/frame.py

```diff
@@ -327,6 +327,68 @@ def notna(self):
         """
         return self.notnull()
 
+    def interleave_columns(self):
+        """
+       Interleave Series columns of a table into a single column.
+
+        Converts the column major table `cols` into a row major column.
+        Parameters
+        ----------
+        cols : input Table containing columns to interleave.
+
+        Example
+        -------
+        >>> df = DataFrame([['A1', 'A2', 'A3'], ['B1', 'B2', 'B3']])
+        >>> df
+        0    [A1, A2, A3]
+        1    [B1, B2, B3]
+        >>> df.interleave_columns()
+        0    A1
+        1    B1
+        2    A2
+        3    B2
+        4    A3
+        5    B3
+
+        Returns
+        -------
+        The interleaved columns as a single column
+        """
+        result = self._constructor_sliced(
+            libcudfxx.reshape.interleave_columns(self)
+        )
+        result._copy_categories(self)
```

Comment:
This doesn't work. This would just copy the categories from the first column of self where if the second column had categories not in the first column we'd silently end up with corrupted data.

Reply:
Connected offline, going to just throw if categorical for the time being